File: Byte&Buy.AngularSPA/src/app/core/dto/offers/common/offer-browser-item-response.ts
import { Guid } from "guid-typescript";
import { ImageResponse } from "../../image/image-response";
import { OfferStatus } from "../enum/offer-status";

export interface OfferBrowserItemResponse{
    id: Guid;
    image: ImageResponse,
    title: string,
    condition: string,
    category: string,
    city: string;
    status: OfferStatus;
    postalCity: string;
    postalCode: string;
    isCompanyOffer: boolean;
    type: 'sale' | 'rent';
}

File: Byte&Buy.Backend/src/ByteBuy.Core/DTO/Public/Offer/Common/OfferBrowserItemResponse.cs
using ByteBuy.Core.DTO.Public.Image;
using ByteBuy.Core.DTO.Public.Offer.RentOffer;
using ByteBuy.Core.DTO.Public.Offer.SaleOffer;
using System.Text.Json.Serialization;

namespace ByteBuy.Core.DTO.Public.Offer.Common;

[JsonPolymorphic(TypeDiscriminatorPropertyName = "type")]
[JsonDerivedType(typeof(SaleBrowserItemResponse), "sale")]
[JsonDerivedType(typeof(RentBrowserItemResponse), "rent")]
public abstract record OfferBrowserItemResponse
{
    public Guid Id { get; init; }
    public ImageResponse Image { get; init; } = null!;
    public string Title { get; init; } = null!;
    public string Condition { get; init; } = null!;
    public string Category { get; init; } = null!;
    public string City { get; init; } = null!;
    public string PostalCity { get; init; } = null!;
    public string PostalCode { get; init; } = null!;
    public bool IsCompanyOffer { get; init; }
}
